File: 2015/day01/1.js
const fs = require('fs');

// Read input.txt
let inp = fs.readFileSync('input.txt', 'utf8');

// Part 1
let curr_floor = 0;
for (let i = 0; i < inp.length; i++) {
    let char = inp[i];
    if (char === '(') {
        curr_floor += 1;
    } else if (char === ')') {
        curr_floor -= 1;
    }
}
console.log(curr_floor);

// Part 2
curr_floor = 0;
let index;
for (index = 0; index < inp.length; index++) {
    let char = inp[index];
    if (char === '(') {
        curr_floor += 1;
    } else if (char === ')') {
        curr_floor -= 1;
    }
    if (curr_floor < 0) {
        break;
    }
}
console.log(index + 1);

File: 2015/day01/1.rs
use std::fs;

fn main() -> std::io::Result<()> {
    let inp = fs::read_to_string("input.txt")?;

    // Part 1
    let mut curr_floor = 0;
    for c in inp.chars() {
        if c == '(' {
            curr_floor += 1;
        } else if c == ')' {
            curr_floor -= 1;
        }
    }
    println!("{}", curr_floor);

    // Part 2
    let mut curr_floor = 0;
    for (index, c) in inp.chars().enumerate() {
        if c == '(' {
            curr_floor += 1;
        } else if c == ')' {
            curr_floor -= 1;
        }
        if curr_floor < 0 {
            println!("{}", index + 1);
            break;
        }
    }

    Ok(())
}
